synapse/storage/databases/main/registration.py

```diff
@@ -443,6 +443,26 @@ def f(txn):
 
         return await self.db_pool.runInteraction("get_users_by_id_case_insensitive", f)
 
+    async def record_user_external_id(
```

Comment:
this is moving from `RegistrationStore` to `RegistrationWorkerStore`. There's no caching on this table so I think it's safe to move.